patches/net/minecraft/server/level/ServerPlayerGameMode.java.patch

```diff
@@ -60,16 +60,18 @@
 -                    boolean flag = this.player.hasCorrectToolForDrops(blockstate);
 +                    boolean flag1 = blockstate.canHarvestBlock(this.level, p_9281_, this.player); // previously player.hasCorrectToolForDrops(blockstate)
```

Comment:
This patch is also incorrect here, in the vanilla logic the state passed to `hasCorrectToolForDrops` is the output of `playerWillDestroy`.  To resolve this we'll have to diverge the creative/survival methods a bit.  Honestly?  We should just remove the `playerWillDestroy` call from `onDestroyedByPlayer` and have it be purely the block-removal logic.

Reply:
Round 2 electric boogaloo. I moved things around and lets see if this is better. Gametest still passes so should be goo there.